client/cmdhfmf.c

```diff
@@ -348,25 +348,23 @@ int GetHFMF14AUID(uint8_t *uid) {
 	iso14a_card_select_t card;
 	memcpy(&card, (iso14a_card_select_t *)resp.d.asBytes, sizeof(iso14a_card_select_t));
 	memcpy(uid, card.uid, card.uidlen * sizeof(uint8_t));
+	*uidlen=card.uidlen;
 	return 1;
 }
 
 char * GenerateFilename(const char *prefix, const char *suffix){
-	uint8_t uid[4] = {0,0,0,0};	
-	int len=0;
+	uint8_t uid[8] = {0,0,0,0,0,0,0,0};
+	int uidlen=0;
 	char * fptr = malloc (sizeof (char) * (strlen(prefix) + strlen(suffix)) + sizeof(uid)*2 + 1);

```

Comment:
do consider clearning the allocated memory with 0x00 before using it.

Reply:
Since the dynamically allocated heap memory will immediately be filled with string, I guess it won't cause any bugs even it is not zero-filled. I agree it must be zero filled/use calloc() if handling binary data.

Reply:
the zerofilling is more to ensure the zero-terminator in the string,  than it is of wrong uid-bytes for name :)